src/utils/self_consistent_field.jl

```diff
@@ -11,15 +11,20 @@ electrons, `ρ` is the initial density, e.g. constructed via a SAD guess.
 for diagonalisation. Possible `algorithm`s are `:scf_nlsolve` or `:scf_damped`.
 """
 function self_consistent_field(ham::Hamiltonian, n_bands::Int, n_electrons::Int;
-                               ρ=nothing, tol=1e-6,
+                               ρ=nothing, tol=1e-6, T=0, smearing=nothing,
                                lobpcg_prec=PreconditionerKinetic(ham, α=0.1),
                                max_iter=100, algorithm=:scf_nlsolve, kwargs...)
-    function compute_occupation(basis, energies, Psi)
-        occupation_zero_temperature(basis, energies, Psi, n_electrons)
-    end
-    if ρ === nothing
-        ρ = guess_hcore(ham, n_bands, compute_occupation, lobpcg_prec=lobpcg_prec)
+    if smearing === nothing
+        compute_occupation(basis, energies, Psi) = occupation_step(basis, energies, Psi,
+                                                                   n_electrons)
+    else
+        function compute_occupation(basis, energies, Psi)
+            occupation_temperature(basis, energies, Psi, n_electrons, T, smearing=smearing)
+        end
     end
```

Comment:
Hm, I get 
`WARNING: Method definition (::getfield(DFTK, Symbol("#compute_occupation#188")){T, smearing, n_electrons})(Any, Any, Any) in module DFTK at /home/antoine/.julia/dev/DFTK/src/utils/self_consistent_field.jl:18 overwritten at /home/antoine/.julia/dev/DFTK/src/utils/self_consistent_field.jl:22.
`
from this

Reply:
WTF: https://discourse.julialang.org/t/conditional-function-definition/27348

Reply:
Interesting ... weird one for sure.